File: src/main/java/br/com/jiratorio/service/ProjectService.java
package br.com.jiratorio.service;

import br.com.jiratorio.domain.jira.BoardStatusList;
import br.com.jiratorio.domain.jira.JiraProject;
import java.util.List;

public interface ProjectService {

    List<JiraProject> findAllJiraProject();

    List<BoardStatusList> findStatusFromProject(Long projectId);

    JiraProject findById(Long projectId);

}

File: src/main/java/br/com/jiratorio/service/ProjectService.kt
package br.com.jiratorio.service

import br.com.jiratorio.domain.jira.JiraProjectDetails
import br.com.jiratorio.domain.jira.JiraProject

interface ProjectService {

    fun findAllJiraProject(): List<JiraProject>

    fun findStatusesByBoardId(boardId: Long): Set<String>

    fun findById(projectId: Long): JiraProjectDetails

}
